psi4/driver/driver_cbs.py

```diff
@@ -1497,11 +1228,577 @@ def _cbs_gufunc(func, total_method_name, **kwargs):
+
+        if data['metadata']:
+            if data['metadata'][0]["wfn"] not in VARH.keys():
+                raise ValidationError(
+                    """Requested SCF method '%s' is not recognized. Add it to VARH in driver_cbs.py to proceed.""" %
+                    (metadata[0]["wfn"]))
+
+            if len(self.metadata) > 1:
+                for delta in self.metadata[1:]:
+                    if delta["wfn"] not in VARH.keys():
+                        raise ValidationError(
+                            f"""Requested higher {delta["treatment"]} method '{delta["wfn"]}' is not recognized. Add it to VARH in driver_cbs.py to proceed."""
+                        )
+                    if delta["wfn_lo"] not in VARH.keys():
+                        raise ValidationError(
+                            f"""Requested lesser {delta["treament"]} method '{delta["wfn_lo"]}' is not recognized. Add it to VARH in driver_cbs.py to proceed."""
+                        )
+
+            self.cbsrec, self.compute_list, self.trove = _build_cbs_compute(self.metameta, self.metadata)
+
+            for job in self.compute_list:
+                keywords = copy.deepcopy(self.metameta['kwargs']['keywords'])
+                if job["f_options"] is not False:
+                    stage_keywords = dict(job["f_options"].items())
+                    keywords = {**keywords, **stage_keywords}
+                task = AtomicComputer(
+                    **{
+                        "molecule": self.molecule,
+                        "driver": self.driver,
+                        "method": job["f_wfn"],
+                        "basis": job["f_basis"],
+                        "keywords": keywords or {},
+                    })
+                self.task_list.append(task)
+
+                # logger.debug("TASK\n" + pp.pformat(task.dict()))
+
+    def build_tasks(self, obj, **kwargs):
+        # permanently a dummy function
+        pass
+
+    def plan(self):
+        # uncalled function
+        return [t.plan() for t in self.task_list]
+
+    def compute(self, client: Optional["FractalClient"] = None):
+        label = self.metameta['label']
+        instructions = "\n" + p4util.banner(f" CBS Computations{':' + label if label else ''} ",
+                                            strNotOutfile=True) + "\n"
+        logger.debug(instructions)
+        core.print_out(instructions)
+
+        with p4util.hold_options_state():
+            for t in reversed(self.task_list):
+                t.compute(client=client)
+
+    def _prepare_results(self, client: Optional["FractalClient"] = None):
+        results_list = [x.get_results(client=client) for x in self.task_list]
+
+        modules = [getattr(v.provenance, "module", None) for v in results_list]
+        if self.driver != "energy" and len(set(modules)) == 2 and modules.count("scf") == len(modules) / 2:
+            # signature of "MP2 GRAD" - "HF GRAD" implementation detail
```

Comment:
Can you elaborate?

Reply:
added "avoid having post-scf single-method gradients/Hessians show up as "(mixed)" module just because an outright HF call in the jobs list"